--- yumdownloader.orig	2013-11-15 00:01:02.000000000 +0400
+++ yumdownloader	2015-05-21 12:48:25.877327687 +0300
@@ -182,7 +182,7 @@
 
             if not installable: # doing one at a time, apart from groups
                 self.logger.error('No Match for argument %s' % pkg)
-                continue
+                sys.exit(1)
             for newpkg in installable:
                 toActOn.extend(_best_convert_pkg2srcpkgs(self, opts, newpkg))
             if toActOn:
@@ -208,7 +208,12 @@
                 self.tsInfo.addInstall(po)
                 self.localPackages.append(po)
             # Resolve dependencies
-            self.resolveDeps()
+            result, errmsgs = self.resolveDeps()
+            if result == 1:
+                self.logger.error('Dependency resolution failed')
+                for msg in errmsgs:
+                    self.logger.error(msg)
+                sys.exit(1)
             # Add newly added packages to the toDownload list
             for pkg in self.tsInfo.getMembers():
                 if not pkg in toDownload:
